vars/neoDeploy.groovy

```diff
@@ -216,7 +231,73 @@ void call(parameters = [:]) {
                 lock("$STEP_NAME:${neoCommandHelper.resourceLock()}") {
                     deploy(script, configuration, neoCommandHelper, configuration.dockerImage, deployMode)
                 }
+                inValidateCache(configuration)
+            }
+        }
+    }
+}
+
+private inValidateCache(configuration){
+    if(configuration.neo.invalidateCache == true) {
+        def account = configuration.neo.account
+        def host = configuration.neo.host
+
+        withCredentials([usernamePassword(
+            credentialsId: configuration.neo.oauthCredentialId,
+            passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+            usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+            def bearerTokenResponse = sh(
+                script: """#!/bin/bash
+                   curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+           \"https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read\"
+                """,
+                returnStdout: true
+            )
+            def fetchBearerTokenJson = readJSON  text: bearerTokenResponse
+
+            def bearerToken = fetchBearerTokenJson.access_token
+
+            echo "Retrieved bearer token."
+
+            def xcsrfTokenResponse = sh(
+                script: """#!/bin/bash
+                               curl -i -L \
+                               -c 'cookies.jar' \
+                               -H 'X-CSRF-Token: Fetch' \
+                               -H "Authorization: Bearer ${bearerToken}" \
+                            \"https://sandboxportal-${account}.${host}/fiori/api/v1/csrf\"
+                    """, returnStdout: true)
+
+            def xcsrfTokenHeaderMatcher=xcsrfTokenResponse =~ /(?m)^X-CSRF-Token: ([0-9A-Z]*)$/
+            def xcsrfToken = xcsrfTokenHeaderMatcher[0][1]
+
+            def siteId = configuration.neo.siteId ? "${configuration.neo.siteId}" : ""
+            if(siteId ==""){
+                echo "Using the default value of siteId from Portal service and invalidating the cache."
             }
+            else{
+                echo "Invalidating the cache for site Id : ${siteId}."
+            }
+                def statusCode = sh(
+                    script: """#!/bin/bash
+                       curl -X POST -L \
+                       -b 'cookies.jar'  \
+                       -H "X-CSRF-Token: ${xcsrfToken}" \
+                       -H "Authorization: Bearer ${bearerToken}" \
+                       -d "{\"siteId\":${siteId}}" \
+                       -so /dev/null \
+                       -w '%{response_code}' \
+                       \"https://sandboxportal-${account}.${host}/fiori/v1/operations/invalidateCache\"
+                  """,
+                    returnStdout: true)
+
+                if (siteId=="" && statusCode == "500") {
```

Comment:
What should happen with other status codes normally indicating an error (`503`, `4xx`)?

Reply:
The main reason to add this check is to make the user easier to understand the issue that `siteId` if not set in the config file, it should be set as default in the portal service, otherwise, the pipeline fails with error code 500. In other cases obviously, it should fail.

Reply:
> In other cases obviously, it should fail.

As far as I can see we will not fail for other http status codes in the 4xx and 5xx range. `curl` itself does not fail in the sense of providing an return code `!=0` when an http 4xx or 5xx happens (unless `--fail` is specified). I _think_ we will get the statusCode and the code flow continues. It is IMO our duty to handle other status codes. Did you test the behaviour in the real world e.g. by providing an url to an non existing endpoint (... should trigger `404`) or by providing invalid credentials (... should trigger `401`) ?

Reply:
Yes, you are right, I have just tested, without `--fail,` the pipeline continues without aborting but fails somewhere which does not give proper reason for the failure.